### PR TITLE
Make sure copier log data transfer so far when receiving a signal.

### PR DESCRIPTION
**Pull request by petterreinholdtsen**

### CODE REVIEW

File: libexec/copier/copier.c [hunk 1]
```diff
@@ -698,7 +719,14 @@ main(int argc, char *argv[])
 	if (getuid() == 0 || geteuid() == 0)
 		WARNING("running as root");
 
+	signal(SIGINT, signal_handler);
+	signal(SIGTERM, signal_handler);
+
 	if (tsdfx_copier(argv[0], argv[1], maxsize) != 0)
```
Comment (dag-erling) [inline libexec/copier/copier.c]: This part is pointless unless you are also going to resend the signal (which is sometimes done so the parent is informed of the fact that the child was terminated by a signal, even though it was caught and handled gracefully)

**Comment by dag-erling** [comment]
`tsdfx_log_complete()` should also be modified to log a different message depending on what happened: successfully copied, early termination due to `-m`, I/O error or signal.

File: libexec/copier/copier.c [hunk 2]
```diff
@@ -698,7 +719,14 @@ main(int argc, char *argv[])
...
-		exit(1);
-	exit(0);
+		retval = 1;
+
+	signal(SIGTERM, SIG_DFL);
```
Comment (dag-erling) [inline libexec/copier/copier.c]: This part is pointless unless you are also going to resend the signal (which is sometimes done so the parent is informed of the fact that the child was terminated by a signal, even though it was caught and handled gracefully)

Comment (petterreinholdtsen) [inline libexec/copier/copier.c]: I did not plan to resend the signal, only return with non-zero exit value.  Why is the code pointless?

Comment (dag-erling) [inline libexec/copier/copier.c]: It adds complexity (introducing a temporary variable to hold the exit code) solely for the purpose of resetting the signal handler immediately before exiting, which was no real effect except that there is a one-in-a-zillion chance that the copier may catch a `SIGINT` or `SIGTERM` after the two `signal(2)` calls but before the `exit(2)` call, in which case the parent will be informed that we died (i.e. failed) when in fact it is entirely possible that we completed successfully before catching the signal.

Comment (petterreinholdtsen) [inline libexec/copier/copier.c]: [Dag-Erling Smørgrav]

> It adds complexity (introducing a temporary variable to hold the exit
> code) solely for the purpose of resetting the signal handler
> immediately before exiting, which was no real effect except that there
> is a one-in-a-zillion chance that the copier may catch a `SIGINT` or
> `SIGTERM` after the two `signal(2)` calls but before the `exit(2)`
> call, in which case the parent will be informed that we died
> (i.e. failed) when in fact it is entirely possible that we completed
> successfully before catching the signal.

Right.  I thought it best to report such unexpected signaling to the
parent, but it is not very important.

I've updated the branch with a new version.

## 

Happy hacking
Petter Reinholdtsen

**Comment by petterreinholdtsen** [comment]
I've lost track, should this be part of the next release or not?  Perhaps the 'timetodie' variable should
use the same name as the new name used by the tsdfx main process?

**Comment by dag-erling** [comment]
Yes, I think so. Let me make a few adjustments.

**Comment by dag-erling** [comment]
Replaced by #65

**Pull request closed without merging**